genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go

```diff
@@ -1447,6 +1448,95 @@ func TestAccResourceTfExportSplitFilesAsHCL(t *testing.T) {
+func TestAccResourceExportManagedSitesAsData(t *testing.T) {
+	var (
+		exportTestDir = filepath.Join("..", "..", ".terraform"+uuid.NewString())
+		resourceID    = "export"
+		configPath    = filepath.Join(exportTestDir, defaultTfJSONFile)
+	)
+
+	if err := telephonyProvidersEdgesSite.CheckForDefaultSite("PureCloud Voice - AWS"); err != nil {
+		t.Skipf("failed to get default site %v", err)
+	}
+
+	defer func(path string) {
+		if err := os.RemoveAll(path); err != nil {
+			t.Logf("failed to remove dir %s: %s", path, err)
+		}
+	}(exportTestDir)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { util.TestAccPreCheck(t) },
+		ProviderFactories: provider.GetProviderFactories(providerResources, providerDataSources),
+		Steps: []resource.TestStep{
+			{
+				Config: generateTfExportByIncludeFilterResources(
+					resourceID,
+					exportTestDir,
+					util.FalseValue, // include_state_file
+					[]string{ // include_filter_resources
+						strconv.Quote("genesyscloud_telephony_providers_edges_site"),
+					},
+					util.FalseValue, // export_as_hcl
+					util.FalseValue,
+					[]string{},
+				),
+				Check: resource.ComposeTestCheckFunc(
+					validateExportManagedSitesAsData(configPath),
+				),
+			},
+		},
+	})
+}
+
+// validateExportManagedSitesAsData verifies that the default managed site 'PureCloud Voice - AWS' is exported as a data source
+func validateExportManagedSitesAsData(filename string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		// Check if the file exists
+		_, err := os.Stat(filename)
+		if err != nil {
+			return fmt.Errorf("failed to find file %s", filename)
+		}
+
+		// Load the JSON content of the export file
+		log.Println("Loading export config into map variable")
+		exportData, err := loadJsonFileToMap(filename)
+		if err != nil {
+			return err
+		}
+		log.Println("Successfully loaded export config into map variable ")
+
+		// Check if data sources exist in the exported data
+		if data, ok := exportData["data"].(map[string]interface{}); ok {
+			sites, ok := data["genesyscloud_telephony_providers_edges_site"].(map[string]interface{})
+			if !ok {
+				return fmt.Errorf("no data sources exported for genesyscloud_telephony_providers_edges_site")
+			}
+
+			log.Println("Checking that managed site with name 'PureCloud Voice - AWS' is exported as data source")
+
+			// Validate each site's name
+			for siteID, site := range sites {
+				siteAttributes, ok := site.(map[string]interface{})
+				if !ok {
+					return fmt.Errorf("unexpected structure for site %s", siteID)
+				}
+
+				siteName, _ := siteAttributes["name"].(string)
+				if siteName == "PureCloud Voice - AWS" {
+					log.Printf("Site %s with name 'PureCloud Voice - AWS' is correctly exported as data source", siteID)
+					return nil
+				}
+			}
+			return fmt.Errorf("site with name 'PureCloud Voice - AWS' was not exported as data source")
+		} else {
```

Comment:
Also better to pass the const value into this function and use it where 'PureCloud Voice - AWS' is mentioned above